src/main/java/com/axis/jenkins/plugins/eiffel/eiffelbroadcaster/BuildWithEiffelLinksAction.java

```diff
@@ -0,0 +1,279 @@
+ * or more Eiffel links. The endpoint is $JOB_URL/eiffel/build and should behave as the regular
+ * $JOB_URL/build endpoint, i.e. you can supply parameters by posting a form with a JSON string.
+ * Additionally, it requires a (possibly empty) list of Eiffel links. Example (payload lacking URL
+ * encoding to improve readability):
+ * <pre>
+ * POST $JOB_URL/eiffel/build
+ * Content-Type: application/x-www-form-urlencoded
+ *
+ * json={"eiffellinks": [{"target": "662b3813-bef4-4588-bf75-ffaead24a6d5", "type": "CAUSE"}], "parameter": [{"name": "PARAM_NAME", "value": "param value"}]}
+ * </pre>
+ * The Eiffel links, if any, will be passed to the build as an {@link EiffelCause} cause.
+ * That cause will be used when piecing together the
+ * {@link com.axis.jenkins.plugins.eiffel.eiffelbroadcaster.eiffel.EiffelActivityTriggeredEvent}
+ * which gets captured in the {@link Run}'s {@link EiffelActivityAction}.
+ */
+public class BuildWithEiffelLinksAction<
+        JobT extends Job<JobT, RunT> & ParameterizedJobMixIn.ParameterizedJob<JobT, RunT>,
+        RunT extends Run<JobT, RunT> & Queue.Executable>
+        implements Action {
+    /** The form parameter that holds the Eiffel links. */
+    public static final String FORM_PARAM_EIFFELLINKS = "eiffellinks";
+
+    /** The form parameter that holds the build parameters. */
+    public static final String FORM_PARAM_PARAMETERS = "parameter";
+
+    /**
+     * The immediate suffix to the URL of the {@link Job}, i.e. the URLs served by this action will have
+     * the form $JOB_URL/$URL_SUFFIX.
+     */
+    public static final String URL_SUFFIX = "eiffel";
+
+    private JobT job;
+
+    public BuildWithEiffelLinksAction(JobT job) {
+        this.job = job;
+    }
+
+    /**
+     * Responds to a /build request by parsing the posted form and transforming the Eiffel links
+     * and (optionally) build parameters into actions that'll get passed to the new build.
+     */
+    @RequirePOST
+    public void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParameter TimeDuration delay)
+            throws IOException {
+        job.checkPermission(Item.BUILD);
+
+        if (!job.isBuildable()) {
+            throw HttpResponses.error(SC_CONFLICT, new IOException(job.getFullName() + " is not buildable"));
+        }
+
+        if (delay == null) {
+            delay = new TimeDuration(TimeUnit.MILLISECONDS.convert(job.getQuietPeriod(), TimeUnit.SECONDS));
+        }
+
+        List<Action> actions = new ArrayList<>();
+        JSONObject formData = null;
+        try {
+            formData = req.getSubmittedForm();
+        }
+        catch (ServletException e) {
+            throw HttpResponses.error(SC_BAD_REQUEST,
+                    new IllegalArgumentException(
+                            "Missing or invalid contents of \"json\" form field: " + e.toString(), e));
+        }
+        try {
+            ParametersDefinitionProperty pp = (ParametersDefinitionProperty) job.getProperty(
```

Comment:
Not needed. You are already defining the class in the method call.

Reply:
Indeed. I could've sworn I had problems with an unexpected need for a cast but it must've been in some other place.